protocol/protos/NetRemoteDataStreamingService.proto

```diff
@@ -0,0 +1,12 @@
+
+syntax = "proto3";
+
+package Microsoft.Net.Remote.Service;
+
+import "NetRemote.proto";
+import "NetRemoteDataStream.proto";
+
+service NetRemoteDataStreaming
+{
+    rpc DataStreamUpload (stream Microsoft.Net.Remote.DataStream.DataStreamUploadData) returns (Microsoft.Net.Remote.DataStream.DataStreamUploadResult);
```

Comment:
I wasn't planning to for this Upload API since the client is in full control - the server simply reads data until the client stops writing, which is determined by the client. I don't think the client needs to specify a "fixed vs continuous" streaming type to the server since the client will just send what it wants.

For the Download API however, the client would request a streaming type and the server would send data according to that requested type - the client this time would read data until the server stops writing, which again is determined by the client (via the fixed number of blocks requested by the client or via an RPC cancellation by the client in the continuous case).

Reply:
Good point, I guess the sever is a pure sink in this case. Thanks!